src/Core/Utilities/AppHelpers.cs

```diff
@@ -434,6 +434,13 @@ public static bool SetAlternateMainPage(AppOptions appOptions)
                     App.MainPage = new NavigationPage(new CipherAddEditPage(appOptions: appOptions));
                     return true;
                 }
+                if (appOptions.FromPasskeyFramework)
+                {
+                    appOptions.FromPasskeyFramework = false;
+                    var deviceActionService = Bit.Core.Utilities.ServiceContainer.Resolve<IDeviceActionService>();
+                    deviceActionService.ReturnToPasskeyAfterUnlock().FireAndForget();
+                    return true;
+                }
```

Comment:
🤔 Should we wait until `ReturnToPasskeyAfterUnlock` finishes to set `FromPasskeyFramework` to `false`? I mean if the method throws an exception or something like that , is it ok the options state to be reset as it's now? Or should the user be redirected again after the error?

Reply:
@fedemkr I think the App will just crash anyway if something goes wrong at this stage.
And in that scenario I suspect the user can press the Passkey Unlock button which will launch the Activity again with the Intent that has the `FromPasskeyFramework="true"`

Reply:
Perfect then, thanks for the explanation 😄